opentelemetry-sdk/src/resource/process.rs

```diff
@@ -25,13 +24,10 @@ impl ResourceDetector for ProcessResourceDetector {
         let arguments = args_os();
         let cmd_arg_val = arguments
             .into_iter()
-            .map(|arg| Cow::from(arg.to_string_lossy().into_owned()))
-            .collect::<Vec<Cow<'_, str>>>();
+            .map(|arg| arg.to_string_lossy().into_owned())
+            .collect::<Vec<String>>();
         Resource::new(vec![
-            KeyValue::new(
-                "process.command_args",
-                Value::Array(Array::String(cmd_arg_val)),
-            ),
+            KeyValue::new("process.command_args", Value::Array(cmd_arg_val.into())),
```

Comment:
For example here we first allocate a `Vec<String>` and then `cmd_arg_val.into()` will allocate a new `Vec<StringValue>`, redoing the outer allocation.

Reply:
Agreed